File: src/models/errors/security.ts
import { decodeSecurityModal } from "~/decoders/security-modal";
import type { SecurityModal } from "../security-modal";

export class SecurityError extends Error {
  public handle: SecurityModal;

  constructor (authentication: any, identity: any, initialUsername?: string) {
    super("You're asked to custom your security methods");
    this.name = "SecurityError";
    this.handle = decodeSecurityModal(authentication, identity, initialUsername);
  }
}

export class SecuritySourceTooLongError extends Error {
  constructor (limit: number) {
    super(`The source name is too long, limited to ${limit} characters`);
    this.name = "SecuritySourceTooLongError";
  }
}

File: library/src/commonMain/kotlin/models/errors/security.kt
package models.errors

import api.private.IdentifyResponse
import decoders.decodeSecurityModal
import kotlinx.serialization.json.JsonObject
import models.SecurityModal

class SecurityError (authentication: JsonObject, identity: IdentifyResponse, initialUsername: String? = null) : Exception("You're asked to custom your security methods") {
    val handle: SecurityModal = decodeSecurityModal(authentication, identity, initialUsername)
}

class SecuritySourceTooLongError (limit: Int) : Exception("The source name is too long, limited to $limit characters")